### ISSUE TITLE
Validate without running a build

### ISSUE THREAD

**Issue by nchlswhttkr** [nchlswhttkr/barhack#1]
So it turns Buildkite has published a [JSON schema for pipeline files](https://buildkite.com/changelog/36-autocomplete-for-your-pipeline-dot-yml-files).

It looks like I can probably use this, so validation can be done without running a build.

```sh
curl -H "Content-Type: plain/text" -X POST "https://barhack.nchlswhttkr.com/lint" -d '<pipeline-file'
# {"status": "PASSED"}
```

I could easily keep the old `/lint` route, maybe rename it to something like `/lint-within-build`.

The new linting route could also be public, because it isn't spinning off builds with each call.

**Comment by nchlswhttkr** [comment]
I could probably also the list of errors with this one as well!

**Comment by nchlswhttkr** [comment]
Implemented with a8e239e3
